pagefaultgames/pokerogue
# [QoL] Making the transfer button show which pokemon can have items be transferred to them

## What are the changes?
This PR makes it so when you go to transfer items off a pokemon, once you choose an item it will show you whether the other pokemon in your party can accept the item or if they're at max stacks.

## Why am I doing these changes?
At the moment it's a bit of a pain to try and figure out what pokemon can take what items, especially on endless with a carry mon. This change aims to make it easier to tell straight away what items your pokemon can take.

## What did change?
I changed the party-ui-handler to do a check for every pokemon when an item is selected. It sees if the item exists on that pokemon, and if so, if it's at max stacks. If so, it hides the hp related fields and shows the TM field (which has been renamed to a description field), and shows "Able" or "Not able" depending on whether the item can be transferred to a pokemon. If a pokemon is doing the transferring, their description field is blank. Lastly, if you cancel or complete a transfer the description fields go away and the health bars come back

This should be mostly good, but I'm not sure if this is all the functionality we want or if we want more? I'm open to suggestions. Also, I've put this up but may have missed some comments or console logs around - I think I got them all, but not 100% sure

### Screenshots/Videos
Below is a screenshot of me trying to transfer a lum berry from my carry mon (Velursby) to my other mons:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/32db0b71-bca0-4f6b-bd08-8cc68017e2e7)

As you can see, Groudon, Garganacl and Klefki have "Not able" next to their names - this indicates that I can't transfer from Velursby to those pokemon, as they are full on lum berries. 

Here's an example of me trying to transfer black glasses from my carry mon to my other party members:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/85ca8717-eef1-4606-ba0a-2f0083c2c96b)

As you can see, all of the pokemon are able to take the black glasses

Below is a video of this in action:

https://github.com/pagefaultgames/pokerogue/assets/66582645/af16edd4-8e41-4c7a-a695-15168f0bcdc6

As you can see, I go to transfer lum berries from velursby, and groundon, garganacl and klefki are unable to accept it. Going into groudon's items you can see it's got 2 lum berries already. Trying to transfer dragon fangs shows that anyone can take it, and transferring silver powder from conkeados also shows that anyone can take it. Lastly transferring reviver seeds shows that groudon, garganacl and velursby once again can't take it, and trying to transfer to groudon shows the usual error. For all of these transfers, the description label of the pokemon doing the transfer is blank.

Below is a screenshot of the changes for double battles: 

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/4e4a96ca-c0e1-41c0-8074-962705b6312c)

Below is a screenshot of transferring to a pokemon that has fainted:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/86d835ad-6a49-4ab9-8757-6434cd5fcf55)

And here's a screenshot of trying to transfer from a pokemon that's fainted:

![image](https://github.com/pagefaultgames/pokerogue/assets/66582645/92b23aaf-b56e-4656-83b7-3f2601ad45ea)

## How to test the changes?
Download this PR and go to transfer items between pokemon. Try to transfer items to pokemon when they're full (i.e. full on berries, reviver seeds, grip claws etc). You should see the description labels of the pokemon say "able" or "not able" in the transfer screen if they can't fit any more of the transfer item

## Checklist
